src/Microsoft.FeatureManagement/FeatureManager.cs

```diff
@@ -217,6 +263,216 @@ await contextualFilter.EvaluateAsync(context, appContext).ConfigureAwait(false)
+            }
+
+            return GetVariantAsync(feature, null, false, cancellationToken);
+        }
+
+        public ValueTask<Variant> GetVariantAsync(string feature, TargetingContext context, CancellationToken cancellationToken)
+        {
+            if (string.IsNullOrEmpty(feature))
+            {
+                throw new ArgumentNullException(nameof(feature));
+            }
+
+            return GetVariantAsync(feature, context, true, cancellationToken);
+        }
+
+        private async ValueTask<Variant> GetVariantAsync(string feature, TargetingContext context, bool useContext, CancellationToken cancellationToken)
+        {
+            FeatureDefinition featureDefinition = await _featureDefinitionProvider
+                .GetFeatureDefinitionAsync(feature)
+                .ConfigureAwait(false);
+
+            if (featureDefinition == null)
+            {
+                throw new FeatureManagementException(
+                    FeatureManagementError.MissingFeature,
+                    $"The feature declaration for the feature '{feature}' was not found.");
+            }
+
+            if (featureDefinition.Allocation == null)
+            {
+                throw new FeatureManagementException(
+                    FeatureManagementError.MissingAllocation,
+                    $"No allocation is defined for the feature {featureDefinition.Name}");
+            }
+
+            if (!featureDefinition.Variants?.Any() ?? false)
+            {
+                throw new FeatureManagementException(
+                    FeatureManagementError.MissingVariantDefinition,
+                    $"No variants are registered for the feature {feature}");
+            }
+
+            VariantDefinition variantDefinition;
+
+            bool isFeatureEnabled = await IsEnabledAsync(feature, context, true, cancellationToken).ConfigureAwait(false);
+
+            variantDefinition = await GetVariantDefinitionAsync(featureDefinition, context, useContext, isFeatureEnabled, cancellationToken).ConfigureAwait(false);
+
+            if (variantDefinition == null)
+            {
+                return null;
+            }
+
+            IConfigurationSection variantConfiguration = null;
+
+            bool configValueSet = !string.IsNullOrEmpty(variantDefinition.ConfigurationValue);
+            bool configReferenceValueSet = !string.IsNullOrEmpty(variantDefinition.ConfigurationReference);
+
+            if (configValueSet && configReferenceValueSet)
+            {
+                throw new FeatureManagementException(
+                    FeatureManagementError.InvalidVariantConfiguration,
+                    $"Both ConfigurationValue and ConfigurationReference are specified for the variant {variantDefinition.Name} in feature {feature}");
+            }
+            else if (configReferenceValueSet)
+            {
+                if (_configuration == null)
+                {
+                    throw new FeatureManagementException(
```

Comment:
same here about throwing. Perhaps it can just reference empty configuration

Reply:
By empty configuration did you mean creating an empty `VariantConfigurationSection` or something else?

Reply:
Yeah

Reply:
But I lean toward returning null for configuration instead. There is no configuration object in the system and returned object can reflect that.